### DISCUSSION TITLE
Constrain an enum value to be unique (Postgres partial index)

### DISCUSSION THREAD

**Discussion by tn819** [prisma/prisma#17868] [Q&A] [answered]
Hi,

I have a use case where I would like to constrain a specific value of an enum to appear only once, but other enum values do not have a unique constraint. How do I express this in Prisma schema?

See below:

```
model Account {
    id               String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
    type             AccountType
    createdAt        DateTime @default(now())
    updatedAt        DateTime @updatedAt

    @@unique([id, organizationName])
    @@unique([name, organizationName])
}

enum AccountType {
  main
  custom
}
```

In Postgres land, this looks like a partial index.

Sample SQL:
```
CREATE INDEX my_main_account_idx ON Accounts (type)
WHERE (type = 'main');
```
Take a look here: [Partial Indexes documentation](http://www.postgresql.org/docs/current/interactive/indexes-partial.html)

**Comment by arosemena** [comment, marked as answer]
Create an empty migration and do your sample SQL

`npx prisma migrate dev --create-only` 

https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations

**Reply by tn819** [reply]
Thanks! This is indeed the way it seems.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
